### PR TITLE
Editor export

### PR DESCRIPTION
**Pull request by antonburger**
Export button now downloads the map text and JSON under the name given in the new input field.

Has the same master as #41, but should be independent of it and easy to merge/rebase.

### CODE REVIEW

**Comment by mjrmua** [comment]
How do you assign reviews like that?

**Comment by mjrmua** [comment]
A link to the editor from the index.html would be nice

**Comment by samperumal** [comment]
> How do you assign reviews like that?

Top left, on git hub, under Reviewers.

File: src/editor.js [hunk 1]
```diff
@@ -146,4 +156,87 @@ class editor {
...
+            const allGroups = allItems.reduce(function(groups, item) {
+                const group = groups[item.t] || (groups[item.t] = []);
+                group.push(item);
+                return groups;
+            }, {});
+
+            // stringify produces line feeds only, so make the result Windows-friendly
+            return JSON
+                .stringify(allGroups, null, 4)
+                .replace(/\n/g, "\r\n");
```
Comment (samperumal) [inline src/editor.js]: Needs a trailing newline after stringification.

**Pull request merged**